node/src/db_config/persistent_configuration.rs

```diff
@@ -416,7 +416,14 @@ impl PersistentConfiguration for PersistentConfigurationReal {
     }
 
     fn max_block_count(&self) -> Result<u64, PersistentConfigError> {
-        self.simple_get_method(decode_u64, "max_block_count")
+        match self.get("max_block_count") {
```

Comment:
I thought I saw somewhere else that the type of `max_block_count` in `PersistentConfiguration` was `Option<u64>`, and that it's initialized in a new database to `None`. Here, though, it seems non-optional. What am I missing?

Also, I saw a new test for the existing `set_max_block_count()` method, but I didn't see any new tests to drive the new logic here.

Reply:
I'll comment on this in the next review I'm about to open. I did some rework here.